<!DOCTYPE html>
<!--[if lt IE 7 ]> <html class="no-js ie6"> <![endif]-->
<!--[if IE 7 ]>    <html class="no-js ie7"> <![endif]-->
<!--[if IE 8 ]>    <html class="no-js ie8"> <![endif]-->
<!--[if IE 9 ]>    <html class="no-js ie9"> <![endif]-->
<!--[if (gt IE 9)|!(IE)]><!--> <html class="no-js"> <!--<![endif]-->
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,user-scalable=no,initial-scale=1,maximum-scale=1">
    
      
        <title>HyPhy - Hypothesis Testing using Phylogenies - Information, documentation, and news about the HyPhy software package</title>
      
      
        <meta name="description" content="Information, documentation, and news about the HyPhy software package">
      
      
        <link rel="canonical" href="http://veg.github.io/hyphy-site/methods/other/fel-contrast/">
      
      
        <meta name="author" content="VEG">
      
    
    <meta property="og:url" content="http://veg.github.io/hyphy-site/methods/other/fel-contrast/">
    <meta property="og:title" content="HyPhy - Hypothesis Testing using Phylogenies">
    <meta property="og:image" content="http://veg.github.io/hyphy-site/methods/other/fel-contrast//../../../images/logo.svg">
    <meta name="apple-mobile-web-app-title" content="HyPhy - Hypothesis Testing using Phylogenies">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    
      <link rel="apple-touch-icon" href="../../../images/logo.svg">
    
    
    <link rel="shortcut icon" type="image/x-icon" href="{{ base_url }}/img/favicon.ico">
    <link rel="icon" type="image/x-icon" href="{{ base_url }}/img/favicon.ico">
    <style>
      @font-face {
      	font-family: 'Icon';
      	src: url('../../../assets/fonts/icon.eot?52m981');
      	src: url('../../../assets/fonts/icon.eot?#iefix52m981')
               format('embedded-opentype'),
      		   url('../../../assets/fonts/icon.woff?52m981')
               format('woff'),
      		   url('../../../assets/fonts/icon.ttf?52m981')
               format('truetype'),
      		   url('../../../assets/fonts/icon.svg?52m981#icon')
               format('svg');
      	font-weight: normal;
      	font-style: normal;
      }
    </style>
    <link rel="stylesheet" href="../../../assets/stylesheets/application.css">
    <link rel="stylesheet" href="../../../assets/stylesheets/index.css">
		<link rel="stylesheet" href="../../../assets/stylesheets/styles.css">
    <link rel="stylesheet" href="../../../assets/stylesheets/mobile.css">
    
    
      
      
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:400,700|Ubuntu+Mono">
      <style>
        body, input {
          font-family: 'Ubuntu', Helvetica, Arial, sans-serif;
        }
        pre, code {
          font-family: 'Ubuntu Mono', 'Courier New', 'Courier', monospace;
        }
      </style>
    
    
    <script src="../../../assets/javascripts/modernizr.js"></script>
    
  </head>
  
  
  
  <body class=" ">
    
    <div class="backdrop">
      <div class="backdrop-paper"></div>
    </div>
    <input class="toggle" type="checkbox" id="toggle-drawer">
    <input class="toggle" type="checkbox" id="toggle-search">
    <label class="toggle-button overlay" for="toggle-drawer"></label>
    <header class="header">
      <nav aria-label="Header">
  <div class="bar default">
    <div class="button button-menu" role="button" aria-label="Menu">
      <label class="toggle-button icon icon-menu" for="toggle-drawer">
        <span></span>
      </label>
    </div>
    <div class="stretch">
      <div class="header-title">
        <a href="#"><img src="../../../images/header-logo.svg" alt="HyPhy Logo" id="header-logo" /></a>
        <a href="#" id="header-title">HyPhy<br />
        <span class="header-title-light">Hypothesis Testing using Phylogenies</span></a>
      </div>
      
      <div class="mobile-header-title">
        <a href="#"><img src="../../../images/header-logo.svg" alt="HyPhy Logo" id="header-logo" /></a>
        <a href="#">HyPhy</a>
      </div>
    </div>
    <div class="button button-search" role="button" aria-label="Search">
      <label class="toggle-button icon icon-search" title="Search" for="toggle-search"></label>
    </div>
  </div>
  <div class="bar search">
    <div class="button button-close" role="button" aria-label="Close">
      <label class="toggle-button icon icon-back" for="toggle-search"></label>
    </div>
    <div class="stretch">
      <div class="field">
        <input class="query" type="text" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false">
      </div>
    </div>
    <div class="button button-reset" role="button" aria-label="Search">
      <button class="toggle-button icon icon-close" id="reset-search"></button>
    </div>
  </div>
</nav>
    </header>
	 <div class="clear-fix"></div>
    <main class="main">
      
      <div class="drawer">
        <nav aria-label="Navigation">
  
  <a href="../../.." class="project">
    <div class="banner">
      
        <div class="logo">
          <img src="../../../images/logo.svg">
        </div>
      
    </div>
  </a>
  <div class="scrollable">
    <div class="wrapper">
      <div class="toc">
        <ul>
          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Home" href="../../.." >
      Home
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="About" href="../../../about/" >
      About
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Download" href="../../../register/" >
      Download
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->



          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Installation" href="../../../installation/" >
      Installation
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Getting Started" href="../../../getting-started/" >
      Getting Started
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Resources" href="../../../resources/" >
      Resources
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->

  <li>
    <span class="section">Methods</span>
    <ul>

      <!-- Render pages of section -->
      
        <!-- Render sections -->


  <li>
    <a class=""
      title="General Information" href="../../general/" >
      General Information
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Selection" href="../../selection-methods/" >
      Selection
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->

  <li>
    <span class="section">Other</span>
    <ul>

      <!-- Render pages of section -->
      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Evolutionary hypothesis testing" href="../evo/" >
      Evolutionary hypothesis testing
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->



      
    </ul>
  </li>

<!-- Render page link -->

      
    </ul>
  </li>

<!-- Render page link -->

          
            <!-- Render sections -->

  <li>
    <span class="section">Tutorials</span>
    <ul>

      <!-- Render pages of section -->
      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Using HyPhy to Detect Selection" href="../../../tutorials/current-release-tutorial/" >
      Using HyPhy to Detect Selection
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Labeling branches with phylotree" href="../../../tutorials/phylotree/" >
      Labeling branches with phylotree
    </a>

    <!-- Expand active pages -->
    
  </li>


      
    </ul>
  </li>

<!-- Render page link -->

          
        </ul>
      </div>
    </div>
  </div>
</nav>
      </div>
      <article class="article">
        <div class="wrapper">
          <h2 id="fel-contrast">FEL-contrast<a class="headerlink" href="#fel-contrast" title="Permanent link">#</a></h2>
<table>
<thead>
<tr>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td>HyPhy version required</td>
<td>&geq; 2.3.4</td>
</tr>
<tr>
<td>Parallel support</td>
<td>MP and/or MPI</td>
</tr>
<tr>
<td>File path</td>
<td><code>LIB/TemplateBatchFiles/SelectionAnalyses/FEL-contrast.bf</code></td>
</tr>
<tr>
<td>Standard analysis menu</td>
<td><code>Evolutionary Hypothesis Testing &gt; FEL-contrast</code></td>
</tr>
</tbody>
</table>
<h3 id="what-biological-question-is-the-method-designed-to-answer">What biological question is the method designed to answer?<a class="headerlink" href="#what-biological-question-is-the-method-designed-to-answer" title="Permanent link">#</a></h3>
<p>Which sites in a gene may be associated with adaptation to a different environment. You need a tree with branches partitioned (a priori) into two sets: reference and test.</p>
<h3 id="what-are-the-recommended-applications">What are the recommended applications?<a class="headerlink" href="#what-are-the-recommended-applications" title="Permanent link">#</a></h3>
<p>Suppose you have a gene which was sampled from different selective environments.
The specific example for which this tool was developed is evolution of HIV in different
hosts or different compartments (blood vs brain) in the host. Similar situations arise
when the gene is sampled from species living in different environment, eating different food,
having different wavelength eye sensitivity. This division has to be binary, however, so
that any branch in the tree is either in the <strong>reference</strong> environment, or the <strong>test</strong> environment.</p>
<p>FEL-contrast then allows you to examine selective pressures (measured as dN/dS) at each site
in the gene individually, and test whether or not they are different between environments.</p>
<p>Armed with a list of such sites, you could then attempt to explore if evolution at these sites
is associated with adaptation to the environment.</p>
<h3 id="what-is-the-statistical-procedure-and-statistical-test-is-used-to-establish-significance-for-this-method">What is the statistical procedure and statistical test is used to establish significance for this method?<a class="headerlink" href="#what-is-the-statistical-procedure-and-statistical-test-is-used-to-establish-significance-for-this-method" title="Permanent link">#</a></h3>
<p>For each site, three rates are inferred, with other parameters (frequencies, branch lengths) inferred jointly and held at</p>
<ul>
<li>&alpha; : synonymous substitution rate</li>
<li>&beta; <sub>r</sub> : non-synonymous substitution rate along <strong>reference</strong> branches</li>
<li>&beta; <sub>p</sub> : non-synonymous substitution rate along <strong>test</strong> branches</li>
</ul>
<p>Two models are compared using a likelihood ratio test</p>
<ul>
<li>
<p>H<sub>A</sub>: &alpha;, &beta; <sub>r</sub>, and &beta; <sub>t</sub> are inferred by maximum likelihood as free parameters</p>
</li>
<li>
<p>H<sub>0</sub>: The &beta; <sub>r</sub> := &beta; <sub>t</sub> constraint is enforced.</p>
</li>
</ul>
<p>The models are nested and differ by one degree of freedom. p-values are computed using the limit &chi;<sup>2</sup> distribution with one degree of freedom.</p>
<h3 id="how-should-one-interpret-positive-and-negative-test-results">How should one interpret positive and negative test results?<a class="headerlink" href="#how-should-one-interpret-positive-and-negative-test-results" title="Permanent link">#</a></h3>
<p>A significant result at a site means that dN/dS (&beta;/&amp;alpha) is <strong>different</strong> between the two sets of branches, with either an <strong>increase</strong> or a <strong>decrease</strong> on the test branches <em>relative</em> to the reference branches. A significant finding does <strong>not</strong> make any claims about positive (dN/dS &gt; 1) or negative (dN / dS &lt; 1), just that dN/dS differ among sets of branches, i.e., a difference need not change the <strong>mode</strong> of selection.</p>
<p>Negative results do not mean that there is no difference, rather that whatever difference there may be does not rise to the level of statistical significance/</p>
<h3 id="rules-of-thumb-for-when-this-method-is-likely-to-work-well-and-when-it-is-not">Rules of thumb for when this method is likely to work well, and when it is not.<a class="headerlink" href="#rules-of-thumb-for-when-this-method-is-likely-to-work-well-and-when-it-is-not" title="Permanent link">#</a></h3>
<ul>
<li>Generally, you need 10 or more branches in each set to be able to have any statistical power.</li>
<li>Too little divergence is also likely to severely throttle statistical power.</li>
</ul>
<h3 id="example">Example<a class="headerlink" href="#example" title="Permanent link">#</a></h3>
<p>We will analyze HIV-1 env sequences from a transmission pair: sequences are isolated from the putative source individual and the putative recipient individual from the 2005 study by <a href="https://www.ncbi.nlm.nih.gov/pubmed/15858036">Frost et al</a>.</p>
<ol>
<li>
<p>Partition the tree into the source and recipient (here we include the transmission branch with the source sequences), for example as described <a href="../../../resources/#labeling-branches-with-phylotree">here</a>. For convenience, <a href="../../../resources-files/data/HIVtransmission.nex">download a NEXUS</a> file with the tree already partitioned.</p>
</li>
<li>
<p>Run <code>HYPHYMP</code> or <code>HYPHYMPI</code>, select <code>Evolutionary Hypothesis Testing</code> from the menu
of analyses then select <code>Use a FEL method to test which sites in a gene may be associated with adaptation to a different environment.</code> Alternatively, you can supply the path of the file as a command line argument, e.g. (by default <code>/path/to/hyphylib</code> should be <code>/usr/local/lib/hyphy</code>)</p>
<p><code>$HYPHYMP /path/to/hyphylib/TemplateBatchFiles/SelectionAnalyses/FEL-contrast.bf</code></p>
</li>
<li>
<p>Select <code>Universal genetic code</code></p>
</li>
<li>
<p>Input the path to the example file downloaded from the link above</p>
</li>
<li>
<p>Choose <code>SOURCE</code> as the test set</p>
</li>
<li>
<p>Select <code>Yes</code> to include synonymous rate variation</p>
</li>
<li>
<p>Input <code>0.1</code> for the default p-value</p>
</li>
</ol>
<p>The analysis will now run for a few minutes and output the following results</p>
<hr />
<h3 id="branches-to-use-as-the-test-set-in-the-fel-contrast-analysis">Branches to use as the test set in the FEL-contrast analysis<a class="headerlink" href="#branches-to-use-as-the-test-set-in-the-fel-contrast-analysis" title="Permanent link">#</a></h3>
<p>Selected 24 branches to include in FEL calculations: <code>0564_7, 0564_11, 0564_4, Node6, 0564_1, 0564_21, 0564_5, Node11, Node9, Node5, 0564_17, Node4, 0564_13, 0564_15, Node16, 0564_22, 0564_6, Node20, 0564_3, Node19, Node15, Node3, 0564_9, Node2</code></p>
<h3 id="obtaining-branch-lengths-and-nucleotide-substitution-biases-under-the-nucleotide-gtr-model">Obtaining branch lengths and nucleotide substitution biases under the nucleotide GTR model<a class="headerlink" href="#obtaining-branch-lengths-and-nucleotide-substitution-biases-under-the-nucleotide-gtr-model" title="Permanent link">#</a></h3>
<ul>
<li>Log(L) = -5524.85, AIC-c = 11151.77 (51 estimated parameters)</li>
</ul>
<h3 id="obtaining-the-global-omega-estimate-based-on-relative-gtr-branch-lengths-and-nucleotide-substitution-biases">Obtaining the global omega estimate based on relative GTR branch lengths and nucleotide substitution biases<a class="headerlink" href="#obtaining-the-global-omega-estimate-based-on-relative-gtr-branch-lengths-and-nucleotide-substitution-biases" title="Permanent link">#</a></h3>
<ul>
<li>Log(L) = -5436.84, AIC-c = 10991.98 (59 estimated parameters)</li>
<li>non-synonymous/synonymous rate ratio for <em>background</em> =   0.9178</li>
<li>non-synonymous/synonymous rate ratio for <em>test</em> =   0.8293</li>
</ul>
<h3 id="improving-branch-lengths-nucleotide-substitution-biases-and-global-dnds-ratios-under-a-full-codon-model">Improving branch lengths, nucleotide substitution biases, and global dN/dS ratios under a full codon model<a class="headerlink" href="#improving-branch-lengths-nucleotide-substitution-biases-and-global-dnds-ratios-under-a-full-codon-model" title="Permanent link">#</a></h3>
<ul>
<li>Log(L) = -5436.29</li>
<li>non-synonymous/synonymous rate ratio for <em>background</em> =   1.1136</li>
<li>non-synonymous/synonymous rate ratio for <em>test</em> =   0.7748</li>
</ul>
<h3 id="for-partition-1-these-sites-are-significant-at-p-01">For partition 1 these sites are significant at p &lt;=0.1<a class="headerlink" href="#for-partition-1-these-sites-are-significant-at-p-01" title="Permanent link">#</a></h3>
<table>
<thead>
<tr>
<th align="center">Codon</th>
<th align="center">alpha</th>
<th align="center">beta-reference</th>
<th align="center">beta-test</th>
<th align="center">LRT</th>
<th align="center">Difference detected?</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center">4</td>
<td align="center">0.000</td>
<td align="center">22.380</td>
<td align="center">0.000</td>
<td align="center">3.390</td>
<td align="center">Decr. p = 0.0656</td>
</tr>
<tr>
<td align="center">52</td>
<td align="center">0.000</td>
<td align="center">20.982</td>
<td align="center">0.000</td>
<td align="center">3.384</td>
<td align="center">Decr. p = 0.0658</td>
</tr>
<tr>
<td align="center">83</td>
<td align="center">0.000</td>
<td align="center">20.365</td>
<td align="center">0.000</td>
<td align="center">3.389</td>
<td align="center">Decr. p = 0.0656</td>
</tr>
<tr>
<td align="center">118</td>
<td align="center">0.000</td>
<td align="center">17.179</td>
<td align="center">0.000</td>
<td align="center">3.404</td>
<td align="center">Decr. p = 0.0651</td>
</tr>
<tr>
<td align="center">124</td>
<td align="center">0.000</td>
<td align="center">23.346</td>
<td align="center">0.000</td>
<td align="center">3.396</td>
<td align="center">Decr. p = 0.0653</td>
</tr>
<tr>
<td align="center">155</td>
<td align="center">0.000</td>
<td align="center">0.000</td>
<td align="center">64.943</td>
<td align="center">5.045</td>
<td align="center">Incr. p = 0.0247</td>
</tr>
<tr>
<td align="center">187</td>
<td align="center">0.000</td>
<td align="center">20.934</td>
<td align="center">0.000</td>
<td align="center">3.577</td>
<td align="center">Decr. p = 0.0586</td>
</tr>
<tr>
<td align="center">218</td>
<td align="center">0.000</td>
<td align="center">20.825</td>
<td align="center">0.000</td>
<td align="center">3.519</td>
<td align="center">Decr. p = 0.0607</td>
</tr>
<tr>
<td align="center">222</td>
<td align="center">0.000</td>
<td align="center">22.658</td>
<td align="center">0.000</td>
<td align="center">3.459</td>
<td align="center">Decr. p = 0.0629</td>
</tr>
<tr>
<td align="center">224</td>
<td align="center">0.000</td>
<td align="center">25.874</td>
<td align="center">0.000</td>
<td align="center">3.681</td>
<td align="center">Decr. p = 0.0550</td>
</tr>
<tr>
<td align="center">352</td>
<td align="center">0.000</td>
<td align="center">19.420</td>
<td align="center">0.000</td>
<td align="center">3.411</td>
<td align="center">Decr. p = 0.0648</td>
</tr>
<tr>
<td align="center">386</td>
<td align="center">0.000</td>
<td align="center">20.334</td>
<td align="center">0.000</td>
<td align="center">3.387</td>
<td align="center">Decr. p = 0.0657</td>
</tr>
<tr>
<td align="center">417</td>
<td align="center">0.000</td>
<td align="center">21.316</td>
<td align="center">0.000</td>
<td align="center">3.383</td>
<td align="center">Decr. p = 0.0659</td>
</tr>
<tr>
<td align="center">455</td>
<td align="center">0.000</td>
<td align="center">22.010</td>
<td align="center">0.000</td>
<td align="center">3.398</td>
<td align="center">Decr. p = 0.0653</td>
</tr>
<tr>
<td align="center">462</td>
<td align="center">0.000</td>
<td align="center">69.066</td>
<td align="center">10.567</td>
<td align="center">3.860</td>
<td align="center">Decr. p = 0.0494</td>
</tr>
<tr>
<td align="center">466</td>
<td align="center">0.000</td>
<td align="center">55.142</td>
<td align="center">0.000</td>
<td align="center">7.562</td>
<td align="center">Decr. p = 0.0060</td>
</tr>
<tr>
<td align="center">506</td>
<td align="center">0.000</td>
<td align="center">33.154</td>
<td align="center">0.000</td>
<td align="center">3.438</td>
<td align="center">Decr. p = 0.0637</td>
</tr>
<tr>
<td align="center">526</td>
<td align="center">0.000</td>
<td align="center">50.810</td>
<td align="center">5.313</td>
<td align="center">3.351</td>
<td align="center">Decr. p = 0.0672</td>
</tr>
<tr>
<td align="center">533</td>
<td align="center">0.000</td>
<td align="center">21.489</td>
<td align="center">0.000</td>
<td align="center">3.485</td>
<td align="center">Decr. p = 0.0619</td>
</tr>
<tr>
<td align="center">598</td>
<td align="center">0.000</td>
<td align="center">18.103</td>
<td align="center">0.000</td>
<td align="center">3.392</td>
<td align="center">Decr. p = 0.0655</td>
</tr>
<tr>
<td align="center">633</td>
<td align="center">7.019</td>
<td align="center">20.227</td>
<td align="center">0.000</td>
<td align="center">3.393</td>
<td align="center">Decr. p = 0.0655</td>
</tr>
<tr>
<td align="center">748</td>
<td align="center">0.000</td>
<td align="center">36.773</td>
<td align="center">0.000</td>
<td align="center">6.388</td>
<td align="center">Decr. p = 0.0115</td>
</tr>
<tr>
<td align="center">751</td>
<td align="center">0.000</td>
<td align="center">18.447</td>
<td align="center">0.000</td>
<td align="center">3.123</td>
<td align="center">Decr. p = 0.0772</td>
</tr>
<tr>
<td align="center">762</td>
<td align="center">0.000</td>
<td align="center">18.868</td>
<td align="center">0.000</td>
<td align="center">3.402</td>
<td align="center">Decr. p = 0.0651</td>
</tr>
<tr>
<td align="center">788</td>
<td align="center">0.000</td>
<td align="center">26.735</td>
<td align="center">0.000</td>
<td align="center">3.937</td>
<td align="center">Decr. p = 0.0472</td>
</tr>
<tr>
<td align="center">820</td>
<td align="center">0.000</td>
<td align="center">56.371</td>
<td align="center">0.000</td>
<td align="center">9.657</td>
<td align="center">Decr. p = 0.0019</td>
</tr>
<tr>
<td align="center">824</td>
<td align="center">0.000</td>
<td align="center">19.630</td>
<td align="center">0.000</td>
<td align="center">3.604</td>
<td align="center">Decr. p = 0.0576</td>
</tr>
</tbody>
</table>
<h3 id="found-1-sites-with-increased-dnds-in-the-test-branches-relative-to-the-reference-branches-and-26-sites-with-decreased-dnds-selection-at-p-01"><strong> Found <em>1</em> sites with </strong>increased<strong> dN/dS in the test branches relative to the reference branches and <em>26</em> sites with </strong>decreased<strong> dN/dS selection at p &lt;= 0.1</strong><a class="headerlink" href="#found-1-sites-with-increased-dnds-in-the-test-branches-relative-to-the-reference-branches-and-26-sites-with-decreased-dnds-selection-at-p-01" title="Permanent link">#</a></h3>
          
            <footer class="footer">
              <div class="container citation-footer">
  <p>HyPhy is funded jointly by <a
                 href="https://projectreporter.nih.gov/project_info_details.cfm?aid=9102131"
                 target="_blank">MIDAS</a> and <a
                 href="https://projectreporter.nih.gov/project_info_description.cfm?aid=8886417"
                 target="_blank">NIH award R01 GM093939</a></p>
</div>



            </footer>
          
        </div>
      </article>
      <div class="results" role="status" aria-live="polite">
        <div class="scrollable">
          <div class="wrapper">
            <div class="meta"></div>
            <div class="list"></div>
          </div>
        </div>
      </div>
    </main>
    <script>
      var base_url = '../../..';
      var repo_id  = '';
    </script>
    <script src="../../../assets/javascripts/application.js"></script>
    
      <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML"></script>
    
      <script src="../../../assets/mathjaxhelper.js"></script>
    
    
				<!-- Piwik -->
				<script type="text/javascript">
					var _paq = _paq || [];
					_paq.push(["setDomains", ["*.hyphy.org"]]);
					_paq.push(['trackPageView']);
					_paq.push(['enableLinkTracking']);
					(function() {
						var u="//datamonkey.org/piwik/";
						_paq.push(['setTrackerUrl', u+'piwik.php']);
						_paq.push(['setSiteId', 10]);
						var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
						g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'piwik.js'; s.parentNode.insertBefore(g,s);
					})();
				</script>
				<noscript><p><img src="//datamonkey.org/piwik/piwik.php?idsite=10" style="border:0;" alt="" /></p></noscript>
				<!-- End Piwik Code -->
    
    <script charset="utf-8">
      var shiftWindow = function() { scrollBy(0, -65) };
      window.addEventListener("hashchange", shiftWindow);
    </script>
  </body>
</html>