<!DOCTYPE html>
<!--[if lt IE 7 ]> <html class="no-js ie6"> <![endif]-->
<!--[if IE 7 ]>    <html class="no-js ie7"> <![endif]-->
<!--[if IE 8 ]>    <html class="no-js ie8"> <![endif]-->
<!--[if IE 9 ]>    <html class="no-js ie9"> <![endif]-->
<!--[if (gt IE 9)|!(IE)]><!--> <html class="no-js"> <!--<![endif]-->
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,user-scalable=no,initial-scale=1,maximum-scale=1">
    
      
        <title>HyPhy - Hypothesis Testing using Phylogenies - Information, documentation, and news about the HyPhy software package</title>
      
      
        <meta name="description" content="Information, documentation, and news about the HyPhy software package">
      
      
        <link rel="canonical" href="http://veg.github.io/hyphy-site/about/">
      
      
        <meta name="author" content="VEG">
      
    
    <meta property="og:url" content="http://veg.github.io/hyphy-site/about/">
    <meta property="og:title" content="HyPhy - Hypothesis Testing using Phylogenies">
    <meta property="og:image" content="http://veg.github.io/hyphy-site/about//../images/logo.svg">
    <meta name="apple-mobile-web-app-title" content="HyPhy - Hypothesis Testing using Phylogenies">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    
      <link rel="apple-touch-icon" href="../images/logo.svg">
    
    
    <link rel="shortcut icon" type="image/x-icon" href="{{ base_url }}/img/favicon.ico">
    <link rel="icon" type="image/x-icon" href="{{ base_url }}/img/favicon.ico">
    <style>
      @font-face {
      	font-family: 'Icon';
      	src: url('../assets/fonts/icon.eot?52m981');
      	src: url('../assets/fonts/icon.eot?#iefix52m981')
               format('embedded-opentype'),
      		   url('../assets/fonts/icon.woff?52m981')
               format('woff'),
      		   url('../assets/fonts/icon.ttf?52m981')
               format('truetype'),
      		   url('../assets/fonts/icon.svg?52m981#icon')
               format('svg');
      	font-weight: normal;
      	font-style: normal;
      }
    </style>
    <link rel="stylesheet" href="../assets/stylesheets/application.css">
    <link rel="stylesheet" href="../assets/stylesheets/index.css">
		<link rel="stylesheet" href="../assets/stylesheets/styles.css">
    <link rel="stylesheet" href="../assets/stylesheets/mobile.css">
    
    
      
      
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:400,700|Ubuntu+Mono">
      <style>
        body, input {
          font-family: 'Ubuntu', Helvetica, Arial, sans-serif;
        }
        pre, code {
          font-family: 'Ubuntu Mono', 'Courier New', 'Courier', monospace;
        }
      </style>
    
    
    <script src="../assets/javascripts/modernizr.js"></script>
    
  </head>
  
  
  
  <body class=" ">
    
    <div class="backdrop">
      <div class="backdrop-paper"></div>
    </div>
    <input class="toggle" type="checkbox" id="toggle-drawer">
    <input class="toggle" type="checkbox" id="toggle-search">
    <label class="toggle-button overlay" for="toggle-drawer"></label>
    <header class="header">
      <nav aria-label="Header">
  <div class="bar default">
    <div class="button button-menu" role="button" aria-label="Menu">
      <label class="toggle-button icon icon-menu" for="toggle-drawer">
        <span></span>
      </label>
    </div>
    <div class="stretch">
      <div class="header-title">
        <a href="#"><img src="../images/header-logo.svg" alt="HyPhy Logo" id="header-logo" /></a>
        <a href="#" id="header-title">HyPhy<br />
        <span class="header-title-light">Hypothesis Testing using Phylogenies</span></a>
      </div>
      
      <div class="mobile-header-title">
        <a href="#"><img src="../images/header-logo.svg" alt="HyPhy Logo" id="header-logo" /></a>
        <a href="#">HyPhy</a>
      </div>
    </div>
    <div class="button button-search" role="button" aria-label="Search">
      <label class="toggle-button icon icon-search" title="Search" for="toggle-search"></label>
    </div>
  </div>
  <div class="bar search">
    <div class="button button-close" role="button" aria-label="Close">
      <label class="toggle-button icon icon-back" for="toggle-search"></label>
    </div>
    <div class="stretch">
      <div class="field">
        <input class="query" type="text" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false">
      </div>
    </div>
    <div class="button button-reset" role="button" aria-label="Search">
      <button class="toggle-button icon icon-close" id="reset-search"></button>
    </div>
  </div>
</nav>
    </header>
	 <div class="clear-fix"></div>
    <main class="main">
      
      <div class="drawer">
        <nav aria-label="Navigation">
  
  <a href=".." class="project">
    <div class="banner">
      
        <div class="logo">
          <img src="../images/logo.svg">
        </div>
      
    </div>
  </a>
  <div class="scrollable">
    <div class="wrapper">
      <div class="toc">
        <ul>
          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Home" href=".." >
      Home
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class="current"
      title="About" href="./" >
      About
    </a>

    <!-- Expand active pages -->
    

      <!--
        The top-level anchor must be skipped if the article contains a h1
        headline, since it would be redundant to the link to the current page
        that is located just above the anchor. Therefore we directly continue
        with the children of the anchor.
      -->
      
        
      

      <!-- Render anchors of active page -->
      
        <ul>
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Overview" href="#overview">
                    Overview
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Brief history" href="#brief-history">
                    Brief history
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Design philosophy" href="#design-philosophy">
                    Design philosophy
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Features" href="#features">
                    Features
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Example HBL Script" href="#example-hbl-script">
                    Example HBL Script
                  </a>
                </li>
            
          
        </ul>
      
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Download" href="../register/" >
      Download
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->



          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Installation" href="../installation/" >
      Installation
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Getting Started" href="../getting-started/" >
      Getting Started
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Resources" href="../resources/" >
      Resources
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->

  <li>
    <span class="section">Methods</span>
    <ul>

      <!-- Render pages of section -->
      
        <!-- Render sections -->


  <li>
    <a class=""
      title="General Information" href="../methods/general/" >
      General Information
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Selection" href="../methods/selection-methods/" >
      Selection
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->

  <li>
    <span class="section">Other</span>
    <ul>

      <!-- Render pages of section -->
      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Evolutionary hypothesis testing" href="../methods/other/evo/" >
      Evolutionary hypothesis testing
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->



      
    </ul>
  </li>

<!-- Render page link -->

      
    </ul>
  </li>

<!-- Render page link -->

          
            <!-- Render sections -->

  <li>
    <span class="section">Tutorials</span>
    <ul>

      <!-- Render pages of section -->
      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Using HyPhy to Detect Selection" href="../tutorials/current-release-tutorial/" >
      Using HyPhy to Detect Selection
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Labeling branches with phylotree" href="../tutorials/phylotree/" >
      Labeling branches with phylotree
    </a>

    <!-- Expand active pages -->
    
  </li>


      
    </ul>
  </li>

<!-- Render page link -->

          
        </ul>
      </div>
    </div>
  </div>
</nav>
      </div>
      <article class="article">
        <div class="wrapper">
          <h2 id="overview">Overview<a class="headerlink" href="#overview" title="Permanent link">#</a></h2>
<p>HyPhy (<strong>Hy</strong>pothesis Testing using <strong>Phy</strong>logenies) is an open-source software package for the analysis of genetic sequences (in particular the inference of natural selection) using techniques in phylogenetics, molecular evolution, and machine learning. It features a rich scripting language for limitless customization of analyses. Additionally, HyPhy features support for parallel computing environments (via message passing interface). HyPhy has over 10000 registered users and has been cited in over 1800 peer-reviewed publications (<a href="https://scholar.google.com/scholar?hl=en&amp;as_sdt=5,39&amp;cites=17874163875017617061,8555001991860797787&amp;scipsc=&amp;q=&amp;scisbd=1">Google Scholar</a>). </p>
<h2 id="brief-history">Brief history<a class="headerlink" href="#brief-history" title="Permanent link">#</a></h2>
<p>HyPhy grew out of the collaboration between <a href="http://www4.stat.ncsu.edu/~muse/">Spencer Muse</a> and <a href="http://hyphy.org/sergei">Sergei Kosakovsky Pond</a> that commenced in 1997 and continues to this day, with many additional collaborators, including <a href="https://www.schulich.uwo.ca/pathol/people/bios/faculty/poon_art.html">Art Poon</a>, <a href="http://www.vet.cam.ac.uk/directory/sdf22@cam.ac.uk">Simon Frost</a>, <a href="http://stevenweaver.org">Steven Weaver</a>, <a href="http://sjspielman.org">Stephanie Spielman</a>, <a href="https://github.com/nlhepler">Lance Hepler</a>, <a href="https://www.linkedin.com/in/martin-smith-371a7717/">Martin Smith</a>, <a href="https://www.linkedin.com/in/konrad-scheffler-b185943/">Konrad Scheffler</a>, <a href="https://www.linkedin.com/in/wayne-delport-5195b545/">Wayne Delport</a>, <a href="http://profiles.ucsd.edu/benjamin.murrell">Ben Murrell</a>, and <a href="http://id.ucsd.edu/faculty/wertheim.shtml">Joel Wertheim</a>. HyPhy was originally released in 2000, is currently at version 2.3, and is presently undergoing a major rewrite, with version v3.0 planned for release in 2018. </p>
<h2 id="design-philosophy">Design philosophy<a class="headerlink" href="#design-philosophy" title="Permanent link">#</a></h2>
<p>HyPhy was designed to allow the specification and fitting of a broad class of continuous-time discrete-space Markov models of sequence evolution. To implement these models, HyPhy provides its own scripting language - <strong>HBL</strong>, or <strong>HyPhy Batch Language</strong> (see <a href="#example-hbl-script">an example</a>), which can be used to develop custom analyses or modify existing ones. Importantly, it is not necessary to learn (or even be aware of) HBL in order to use HyPhy, as most common models and analyses have been implemented for user convenience. Once a model is defined, it can be fitted to data (using a fixed topology tree), its parameters can be constrained in user-defined ways to test various hypotheses (e.g. is rate1 &gt; rate2), and simulate data from. HyPhy primarily implements <strong>maximum likelihood</strong> methods, but it can also be used to perform some forms of Bayesian inference (e.g. <a href="../methods/selection-methods/#fubar">FUBAR</a>), fit Bayesian graphical models to data, run genetic algorithms to perform complex model selection.</p>
<h2 id="features">Features<a class="headerlink" href="#features" title="Permanent link">#</a></h2>
<ol>
<li>Support for arbitrary sequence data, including nucleotide, amino-acid, codon, binary, count (microsattelite) data, including multiple partitions mixing differen data types. </li>
<li>Complex models of rate variation, including site-to-site, branch-to-branch, hidden markov model (autocorrelated rates), between/within partitions, and co-varion type models.</li>
<li>Fast numerical fitting routines, supporting parallel and distributed execution.</li>
<li>A broad collection of pre-defined evolutionary models. </li>
<li>The ability to specify flexible constraints on model parameters and estimate confidence intervals on MLEs.</li>
<li>Ancestral sequence reconstruction and sampling. </li>
<li>Simulate data from any model that can be defined and fitted in the language.</li>
<li>Apply unique (for this domain) machine learning methods to discover patterns in the data, e.g. genetic algorithms, stochastic context free grammars, Bayesian graphical models.</li>
<li>Script analyses completely in HBL including flow control, I/O, parallelization, etc.</li>
<li>(v2.3) Modern web-applications for interactive result visualization.</li>
</ol>
<h2 id="example-hbl-script">Example HBL Script<a class="headerlink" href="#example-hbl-script" title="Permanent link">#</a></h2>
<div class="code"><pre><span></span><span class="cm">/* </span>
<span class="cm">   This is an example HYPHY Batch File.</span>
<span class="cm">   It reads in a MEGA format nucleotide dataset from data/hiv.nuc.</span>
<span class="cm">   and fits the F81 model using the tree inclded in the file using maximum likelihood.</span>
<span class="cm">   Output is printed out as a Newick Style tree with branch lengths</span>
<span class="cm">   representing the number of expected substitutions per branch.</span>
<span class="cm">*/</span>

<span class="c1">// 1. Read in the data and store the result in a DataSet variable</span>

<span class="nx">DataSet</span>         <span class="nx">nucleotideSequences</span> <span class="o">=</span> <span class="nx">ReadDataFile</span> <span class="p">(</span><span class="s2">&quot;data/hiv.nuc&quot;</span><span class="p">);</span>

<span class="c1">// 2. Filter the data, specifying that all of the data is to be used</span>
<span class="c1">//  and that it is to be treated as nucleotides.*/</span>

<span class="nx">DataSetFilter</span>   <span class="nx">filteredData</span> <span class="o">=</span> <span class="nx">CreateFilter</span> <span class="p">(</span><span class="nx">nucleotideSequences</span><span class="p">,</span><span class="mi">1</span><span class="p">);</span>

<span class="c1">// Collect observed nucleotide frequencies from the filtered data. observedFreqs will</span>
<span class="c1">// store receieve the vector of frequencies. </span>

<span class="nx">HarvestFrequencies</span> <span class="p">(</span><span class="nx">observedFreqs</span><span class="p">,</span> <span class="nx">filteredData</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">,</span> <span class="mi">1</span><span class="p">);</span>

<span class="c1">// Define the F81 substitution matrix. &#39;*&#39; is defined to be -(sum of off-diag row </span>
<span class="c1">// elements); mu is the rate*time parameter </span>

<span class="nx">F81RateMatrix</span> <span class="o">=</span> 
        <span class="p">{{</span><span class="o">*</span><span class="p">,</span><span class="nx">mu</span><span class="p">,</span><span class="nx">mu</span><span class="p">,</span><span class="nx">mu</span><span class="p">}</span>
         <span class="p">{</span><span class="nx">mu</span><span class="p">,</span><span class="o">*</span><span class="p">,</span><span class="nx">mu</span><span class="p">,</span><span class="nx">mu</span><span class="p">}</span>
         <span class="p">{</span><span class="nx">mu</span><span class="p">,</span><span class="nx">mu</span><span class="p">,</span><span class="o">*</span><span class="p">,</span><span class="nx">mu</span><span class="p">}</span>
         <span class="p">{</span><span class="nx">mu</span><span class="p">,</span><span class="nx">mu</span><span class="p">,</span><span class="nx">mu</span><span class="p">,</span><span class="o">*</span><span class="p">}};</span>

<span class="c1">//  Define the F81 models, by combining the substitution matrix with the vector of observed (equilibrium) frequencies.</span>

<span class="nx">Model</span>   <span class="nx">F81</span> <span class="o">=</span> <span class="p">(</span><span class="nx">F81RateMatrix</span><span class="p">,</span> <span class="nx">observedFreqs</span><span class="p">);</span>

<span class="c1">// Now we can define the tree variable, using the tree string read from the data file, </span>
<span class="c1">// and, by default, assigning the last defined model (F81) to all tree branches.</span>

<span class="nx">Tree</span>    <span class="nx">givenTree</span> <span class="o">=</span> <span class="nx">DATAFILE_TREE</span><span class="p">;</span>

<span class="c1">// Since all the likelihood function ingredients (data, tree, equilibrium frequencies)</span>
<span class="c1">// have been defined we are ready to construct the likelihood function.</span>

<span class="nx">LikelihoodFunction</span>  <span class="nx">LF</span> <span class="o">=</span> <span class="p">(</span><span class="nx">filteredData</span><span class="p">,</span> <span class="nx">givenTree</span><span class="p">);</span>

<span class="c1">// Maximize the likelihood function, storing parameter values in the matrix paramValues </span>

<span class="nx">Optimize</span> <span class="p">(</span><span class="nx">paramValues</span><span class="p">,</span> <span class="nx">LF</span><span class="p">);</span>

<span class="c1">// Print the tree with optimal branch lengths to the console. </span>

<span class="nx">fprintf</span>  <span class="p">(</span><span class="nx">stdout</span><span class="p">,</span> <span class="nx">LF</span><span class="p">);</span>
</pre></div>
          
            <footer class="footer">
              <div class="container citation-footer">
  <p>HyPhy is funded jointly by <a
                 href="https://projectreporter.nih.gov/project_info_details.cfm?aid=9102131"
                 target="_blank">MIDAS</a> and <a
                 href="https://projectreporter.nih.gov/project_info_description.cfm?aid=8886417"
                 target="_blank">NIH award R01 GM093939</a></p>
</div>



            </footer>
          
        </div>
      </article>
      <div class="results" role="status" aria-live="polite">
        <div class="scrollable">
          <div class="wrapper">
            <div class="meta"></div>
            <div class="list"></div>
          </div>
        </div>
      </div>
    </main>
    <script>
      var base_url = '..';
      var repo_id  = '';
    </script>
    <script src="../assets/javascripts/application.js"></script>
    
      <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML"></script>
    
      <script src="../assets/mathjaxhelper.js"></script>
    
    
				<!-- Piwik -->
				<script type="text/javascript">
					var _paq = _paq || [];
					_paq.push(["setDomains", ["*.hyphy.org"]]);
					_paq.push(['trackPageView']);
					_paq.push(['enableLinkTracking']);
					(function() {
						var u="//datamonkey.org/piwik/";
						_paq.push(['setTrackerUrl', u+'piwik.php']);
						_paq.push(['setSiteId', 10]);
						var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
						g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'piwik.js'; s.parentNode.insertBefore(g,s);
					})();
				</script>
				<noscript><p><img src="//datamonkey.org/piwik/piwik.php?idsite=10" style="border:0;" alt="" /></p></noscript>
				<!-- End Piwik Code -->
    
    <script charset="utf-8">
      var shiftWindow = function() { scrollBy(0, -65) };
      window.addEventListener("hashchange", shiftWindow);
    </script>
  </body>
</html>