<!DOCTYPE html>
<!--[if lt IE 7 ]> <html class="no-js ie6"> <![endif]-->
<!--[if IE 7 ]>    <html class="no-js ie7"> <![endif]-->
<!--[if IE 8 ]>    <html class="no-js ie8"> <![endif]-->
<!--[if IE 9 ]>    <html class="no-js ie9"> <![endif]-->
<!--[if (gt IE 9)|!(IE)]><!--> <html class="no-js"> <!--<![endif]-->
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,user-scalable=no,initial-scale=1,maximum-scale=1">
    
      
        <title>HyPhy - Hypothesis Testing using Phylogenies - Information, documentation, and news about the HyPhy software package</title>
      
      
        <meta name="description" content="Information, documentation, and news about the HyPhy software package">
      
      
        <link rel="canonical" href="http://veg.github.io/hyphy-site/tutorials/current-release-tutorial/">
      
      
        <meta name="author" content="VEG">
      
    
    <meta property="og:url" content="http://veg.github.io/hyphy-site/tutorials/current-release-tutorial/">
    <meta property="og:title" content="HyPhy - Hypothesis Testing using Phylogenies">
    <meta property="og:image" content="http://veg.github.io/hyphy-site/tutorials/current-release-tutorial//../../images/logo.svg">
    <meta name="apple-mobile-web-app-title" content="HyPhy - Hypothesis Testing using Phylogenies">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
    
      <link rel="apple-touch-icon" href="../../images/logo.svg">
    
    
    <link rel="shortcut icon" type="image/x-icon" href="{{ base_url }}/img/favicon.ico">
    <link rel="icon" type="image/x-icon" href="{{ base_url }}/img/favicon.ico">
    <style>
      @font-face {
      	font-family: 'Icon';
      	src: url('../../assets/fonts/icon.eot?52m981');
      	src: url('../../assets/fonts/icon.eot?#iefix52m981')
               format('embedded-opentype'),
      		   url('../../assets/fonts/icon.woff?52m981')
               format('woff'),
      		   url('../../assets/fonts/icon.ttf?52m981')
               format('truetype'),
      		   url('../../assets/fonts/icon.svg?52m981#icon')
               format('svg');
      	font-weight: normal;
      	font-style: normal;
      }
    </style>
    <link rel="stylesheet" href="../../assets/stylesheets/application.css">
    <link rel="stylesheet" href="../../assets/stylesheets/index.css">
		<link rel="stylesheet" href="../../assets/stylesheets/styles.css">
    <link rel="stylesheet" href="../../assets/stylesheets/mobile.css">
    
    
      
      
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Ubuntu:400,700|Ubuntu+Mono">
      <style>
        body, input {
          font-family: 'Ubuntu', Helvetica, Arial, sans-serif;
        }
        pre, code {
          font-family: 'Ubuntu Mono', 'Courier New', 'Courier', monospace;
        }
      </style>
    
    
    <script src="../../assets/javascripts/modernizr.js"></script>
    
  </head>
  
  
  
  <body class=" ">
    
    <div class="backdrop">
      <div class="backdrop-paper"></div>
    </div>
    <input class="toggle" type="checkbox" id="toggle-drawer">
    <input class="toggle" type="checkbox" id="toggle-search">
    <label class="toggle-button overlay" for="toggle-drawer"></label>
    <header class="header">
      <nav aria-label="Header">
  <div class="bar default">
    <div class="button button-menu" role="button" aria-label="Menu">
      <label class="toggle-button icon icon-menu" for="toggle-drawer">
        <span></span>
      </label>
    </div>
    <div class="stretch">
      <div class="header-title">
        <a href="#"><img src="../../images/header-logo.svg" alt="HyPhy Logo" id="header-logo" /></a>
        <a href="#" id="header-title">HyPhy<br />
        <span class="header-title-light">Hypothesis Testing using Phylogenies</span></a>
      </div>
      
      <div class="mobile-header-title">
        <a href="#"><img src="../../images/header-logo.svg" alt="HyPhy Logo" id="header-logo" /></a>
        <a href="#">HyPhy</a>
      </div>
    </div>
    <div class="button button-search" role="button" aria-label="Search">
      <label class="toggle-button icon icon-search" title="Search" for="toggle-search"></label>
    </div>
  </div>
  <div class="bar search">
    <div class="button button-close" role="button" aria-label="Close">
      <label class="toggle-button icon icon-back" for="toggle-search"></label>
    </div>
    <div class="stretch">
      <div class="field">
        <input class="query" type="text" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false">
      </div>
    </div>
    <div class="button button-reset" role="button" aria-label="Search">
      <button class="toggle-button icon icon-close" id="reset-search"></button>
    </div>
  </div>
</nav>
    </header>
	 <div class="clear-fix"></div>
    <main class="main">
      
      <div class="drawer">
        <nav aria-label="Navigation">
  
  <a href="../.." class="project">
    <div class="banner">
      
        <div class="logo">
          <img src="../../images/logo.svg">
        </div>
      
    </div>
  </a>
  <div class="scrollable">
    <div class="wrapper">
      <div class="toc">
        <ul>
          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Home" href="../.." >
      Home
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="About" href="../../about/" >
      About
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Download" href="../../register/" >
      Download
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->



          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Installation" href="../../installation/" >
      Installation
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Getting Started" href="../../getting-started/" >
      Getting Started
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->


  <li>
    <a class=""
      title="Resources" href="../../resources/" >
      Resources
    </a>

    <!-- Expand active pages -->
    
  </li>


          
            <!-- Render sections -->

  <li>
    <span class="section">Methods</span>
    <ul>

      <!-- Render pages of section -->
      
        <!-- Render sections -->


  <li>
    <a class=""
      title="General Information" href="../../methods/general/" >
      General Information
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Selection" href="../../methods/selection-methods/" >
      Selection
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->

  <li>
    <span class="section">Other</span>
    <ul>

      <!-- Render pages of section -->
      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Evolutionary hypothesis testing" href="../../methods/other/evo/" >
      Evolutionary hypothesis testing
    </a>

    <!-- Expand active pages -->
    
  </li>


      
        <!-- Render sections -->



      
    </ul>
  </li>

<!-- Render page link -->

      
    </ul>
  </li>

<!-- Render page link -->

          
            <!-- Render sections -->

  <li>
    <span class="section">Tutorials</span>
    <ul>

      <!-- Render pages of section -->
      
        <!-- Render sections -->


  <li>
    <a class="current"
      title="Using HyPhy to Detect Selection" href="./" >
      Using HyPhy to Detect Selection
    </a>

    <!-- Expand active pages -->
    

      <!--
        The top-level anchor must be skipped if the article contains a h1
        headline, since it would be redundant to the link to the current page
        that is located just above the anchor. Therefore we directly continue
        with the children of the anchor.
      -->
      
        
      

      <!-- Render anchors of active page -->
      
        <ul>
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Before you begin" href="#before-you-begin">
                    Before you begin
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Preparing labeled phylogenies" href="#preparing-labeled-phylogenies">
                    Preparing labeled phylogenies
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Preparing input data for HyPhy" href="#preparing-input-data-for-hyphy">
                    Preparing input data for HyPhy
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="General Information" href="#general-information">
                    General Information
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Use BUSTED to test for alignment-wide episodic diversifying selection." href="#use-busted-to-test-for-alignment-wide-episodic-diversifying-selection">
                    Use BUSTED to test for alignment-wide episodic diversifying selection.
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Use aBSREL to find lineages which have experienced episodic diversifying selection." href="#use-absrel-to-find-lineages-which-have-experienced-episodic-diversifying-selection">
                    Use aBSREL to find lineages which have experienced episodic diversifying selection.
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Use FEL to find sites which have experienced pervasive diversifying selection." href="#use-fel-to-find-sites-which-have-experienced-pervasive-diversifying-selection">
                    Use FEL to find sites which have experienced pervasive diversifying selection.
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Use MEME to find sites which have experienced pervasive diversifying selection." href="#use-meme-to-find-sites-which-have-experienced-pervasive-diversifying-selection">
                    Use MEME to find sites which have experienced pervasive diversifying selection.
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Use SLAC to find sites which have experienced pervasive diversifying selection." href="#use-slac-to-find-sites-which-have-experienced-pervasive-diversifying-selection">
                    Use SLAC to find sites which have experienced pervasive diversifying selection.
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Use FUBAR to find sites which have experienced pervasive diversifying selection." href="#use-fubar-to-find-sites-which-have-experienced-pervasive-diversifying-selection">
                    Use FUBAR to find sites which have experienced pervasive diversifying selection.
                  </a>
                </li>
            
          

            <!-- Render anchor -->
            
                <li class="anchor">
                  <a title="Use RELAX to compare selective pressures on different parts of the tree" href="#use-relax-to-compare-selective-pressures-on-different-parts-of-the-tree">
                    Use RELAX to compare selective pressures on different parts of the tree
                  </a>
                </li>
            
          
        </ul>
      
    
  </li>


      
        <!-- Render sections -->


  <li>
    <a class=""
      title="Labeling branches with phylotree" href="../phylotree/" >
      Labeling branches with phylotree
    </a>

    <!-- Expand active pages -->
    
  </li>


      
    </ul>
  </li>

<!-- Render page link -->

          
        </ul>
      </div>
    </div>
  </div>
</nav>
      </div>
      <article class="article">
        <div class="wrapper">
          <h1 id="using-hyphy-to-detect-selection">Using HyPhy to detect selection.<a class="headerlink" href="#using-hyphy-to-detect-selection" title="Permanent link">#</a></h1>
<p>These tutorials outline how to prepare data and execute analyses in HyPhy's suite of methods for detecting natural selection in protein-coding alignments. Specifically, this tutorial explains how to use the <a href="https://github.com/veg/hyphy/releases">current release</a> of HyPhy from the <strong>command line</strong>. </p>
<p>All analyses described here produce a final output JSON-formatted file which can be uploaded to <a href="http://vision.hyphy.org">HyPhy Vision</a> for exploration. You can obtain a description of JSON contents for all analyses <a href="../../resources/json-fields.pdf">here</a>. In addition, each analyses will provide live <a href="https://en.wikipedia.org/wiki/Markdown">Markdown-formatted</a> status indicators to the console while running. </p>
<h3 id="before-you-begin">Before you begin<a class="headerlink" href="#before-you-begin" title="Permanent link">#</a></h3>
<ol>
<li>Install the current release of HyPhy on your computer, as needed, using <a href="../../installation/">these instructions</a>.</li>
<li>This tutorial employs example datasets, available for download as a <a href="https://github.com/veg/hyphy-site/blob/master/docs/tutorials/files/tutorial_data.zip?raw=true">zip file</a>. Unpack this zip file on your machine for use and <strong>remember the absolute path to this directory</strong>. All datasets and output JSON files for this tutorial are in this zip file.</li>
<li>This tutorial assumes you are specifically using the HyPhy executable <code>HYPHYMP</code>. If you have installed a different executable (e.g. <code>HYPHYMPI</code>), you may need to alter some commands.
<!--4. This tutorial uses the interactive HyPhy menu prompt to perform analyses. If you wish to automate many analyses instead of using HyPhy interactively, see the section <a href="./current-release-tutorial/#automating-analyses">Automating Analyses</a> for modified instructions.--></li>
</ol>
<h3 id="preparing-labeled-phylogenies">Preparing labeled phylogenies<a class="headerlink" href="#preparing-labeled-phylogenies" title="Permanent link">#</a></h3>
<p>Several analyses are accept labeled phylogenies to define branch sets for selection testing. Moreover, the method <strong>RELAX</strong> <em>requires</em> a labeled phylogeny to compare selection pressures. To assist in tree labeling, we recommend using our <a href="http://phylotree.hyphy.org">Phylotree Widget</a>. Instructions for using this widget are available <a href="../phylotree/">here</a>.</p>
<h3 id="preparing-input-data-for-hyphy">Preparing input data for HyPhy<a class="headerlink" href="#preparing-input-data-for-hyphy" title="Permanent link">#</a></h3>
<p>All analyses require an alignment and corresponding phylogeny for analysis. There are options for preparing your data:</p>
<ul>
<li>Prepare your data in two separate files with the alignment and phylogeny each. Most standard alignemnt formats are accepted (FASTA, phylip, etc.), and the phylogeny should be newick-formatted.</li>
<li>Prepare your data in a single file containing a FASTA-formatted alignment, beneath which should be a newick-formatted phylogeny.</li>
<li>Prepare your data as a NEXUS file with both a data matrix and tree block. Note that this file type will be necessary for performing a partitioned analysis, where different sites evolve according to different phylogenies (i.e. a recombination-corrected dataset from <a href="../../methods/selection-methods/#gard"><strong>GARD</strong></a>.</li>
</ul>
<p>Each of these choices will trigger a slightly different data-input prompt, as described in the <strong>General Information</strong> section below.</p>
<h3 id="general-information">General Information<a class="headerlink" href="#general-information" title="Permanent link">#</a></h3>
<p>All available selection analyses in HyPhy can be accessed by launching HyPhy from the command line by typing <code>HYPHYMP</code> (or launching <code>HYPHYMPI</code> in an appropriate MPI environment) and entering <strong><code>1</code></strong> to reach the <strong><code>Selection Analyses</code></strong> menu: 
<img alt="HyPhy Selection Analyses Menu" src="../files/images/selection-analyses-menu.png" /></p>
<p>In this menu, launch your desired analyis by issuing its associated number (i.e. launch FEL by entering <strong><code>2</code></strong> upon reaching this menu).</p>
<p>Within each analysis, you will see a series of prompts for providing information. All analyses begin with the following prompts:</p>
<ul>
<li>
<p><strong><code>Choose Genetic Code</code></strong>. Generally, the universal genetic code (<strong><code>1</code></strong>) should be provided here, unless the dataset of interest uses a different NCBI-defined genetic code. For each option, the corresponding NCBI translation table is indicated.</p>
</li>
<li>
<p><strong><code>Select a coding sequence alignment file</code></strong>. This option prompts for the dataset to analyze. Provide the <strong>full path</strong> to the dataset of interest. </p>
<ul>
<li>If you provide a file containing only an alignment, HyPhy will issue a subsequent prompt: <strong><code>Please select a tree file for the data</code></strong>. Supply the full path to your newick-formatted phylogeny here.</li>
<li>If you provide a file containing both an alignment and tree, HyPhy will prompt you to confirm that the tree provided should be used: <strong><code>A tree was found in the data file:...Would you like to use it (y/n)?</code></strong>. Enter <strong><code>y</code></strong> to use this tree, or enter <strong><code>n</code></strong> if a different tree is desired. HyPhy will then prompt for this path.</li>
<li>If you provide a NEXUS file, Hyphy will accept the tree(s) as given and will not issue a subsequent prompt.</li>
</ul>
</li>
</ul>
<h3 id="use-busted-to-test-for-alignment-wide-episodic-diversifying-selection">Use BUSTED to test for alignment-wide episodic diversifying selection.<a class="headerlink" href="#use-busted-to-test-for-alignment-wide-episodic-diversifying-selection" title="Permanent link">#</a></h3>
<blockquote>
<p>See <a href="../../methods/selection-methods/#busted">here</a> for a description of the BUSTED method.</p>
</blockquote>
<p>We will demonstate BUSTED use with an alignment of primate sequences for the KSR2 gene, a kinase suppressor of RAS-2, from <a href="https://elifesciences.org/articles/12469">Enard et al, 2016</a>. This dataset is in the file <code>ksr2.fna</code>.</p>
<p>Launch HyPhy from the command line by typing <code>HYPHYMP</code>. Navigate through the prompt to reach the BUSTED analysis: Enter <strong>1</strong> for "Selection Analyses", and then <strong>5</strong> for "BUSTED". Respond to the following prompts:</p>
<ul>
<li><strong><code>Choose Genetic Code</code></strong>: Enter <code>1</code> to select the Universal genetic code.</li>
<li><strong><code>Select a coding sequence alignment file</code></strong>: Enter the full path to the example dataset, <code>/path/to/tutorial_data/ksr2.fna</code></li>
<li><strong><code>A tree was found in the data file:...Would you like to use it (y/n)?</code></strong>: Enter <code>y</code> to use the provided tree.</li>
<li><strong><code>Choose the set of branches to test for selection</code></strong>: To execute a BUSTED analysis that tests the entire tree for selection, enter option <code>1</code> for <strong>All</strong>. Alternatively, if you wish to test a subset of branches, enter a different option (2,3,4, or other). Note that any labels present in the provided phylogeny will be listed as options in this menu.</li>
</ul>
<p>BUSTED will now run to completion and print markdown-formatted status indicators to screen, indicating the progression of model fits and concluding with the final BUSTED test results:
<img alt="BUSTED markdown output" src="../files/images/busted-markdown.png" /></p>
<p>We therefore find that there is evidence for positive, diversifying selection in this dataset, at P=0.0015.</p>
<h3 id="use-absrel-to-find-lineages-which-have-experienced-episodic-diversifying-selection">Use aBSREL to find lineages which have experienced episodic diversifying selection.<a class="headerlink" href="#use-absrel-to-find-lineages-which-have-experienced-episodic-diversifying-selection" title="Permanent link">#</a></h3>
<blockquote>
<p>See <a href="../../methods/selection-methods/#absrel">here</a> for a description of the aBSREL method.</p>
</blockquote>
<p>We will demonstate aBSREL use with an alignment of HIV-1 envelope protein sequences collected from  epidemiologically-linked donor-recipient transmission pairs, from <a href="http://jvi.asm.org/content/79/10/6523">Frost et al, 2005</a>. This dataset is in the file <code>hiv1_transmission.fna</code>.</p>
<p>Launch HyPhy from the command line by typing <code>HYPHYMP</code>. Navigate through the prompt to reach the aBSREL analysis: Enter <strong>1</strong> for "Selection Analyses", and then <strong>6</strong> for "aBSREL". Respond to the following prompts:</p>
<ul>
<li><strong><code>Choose Genetic Code</code></strong>: Enter <code>1</code> to select the Universal genetic code.</li>
<li><strong><code>Select a coding sequence alignment file</code></strong>: Enter the full path to the example dataset, <code>/path/to/tutorial_data/hiv1_transmission.fna</code></li>
<li><strong><code>A tree was found in the data file:...Would you like to use it (y/n)?</code></strong>: Enter <code>y</code> to use the provided tree.</li>
<li><strong><code>Choose the set of branches to test for selection</code></strong>: To test for selection on each branch of your tree (an "exploratory" analysis that may suffer from low power), enter option <code>1</code> for <strong>All</strong>. Alternatively, if you wish to test for selection only on a subset of branches, enter a different option (2,3,4, or other). Note that any labels present in the provided phylogeny will be listed as options in this menu.</li>
</ul>
<p>aBSREL will now run to completion and print markdown-formatted status indicators to screen, indicating the progression of model fits and concluding with the final aBSREL test results (abbreviated output shown here with final result only):
<img alt="aBSREL markdown output" src="../files/images/absrel-markdown.png" /></p>
<p>We therefore find that there is evidence for episodic diversifying selection in this dataset along three branches, after applying the Bonferroni-Holm procedure to control family-wise error rates.         </p>
<h3 id="use-fel-to-find-sites-which-have-experienced-pervasive-diversifying-selection">Use FEL to find sites which have experienced pervasive diversifying selection.<a class="headerlink" href="#use-fel-to-find-sites-which-have-experienced-pervasive-diversifying-selection" title="Permanent link">#</a></h3>
<blockquote>
<p>See <a href="../../methods/selection-methods/#fel">here</a> for a description of the FEL method.</p>
</blockquote>
<p>We will demonstate FEL use with an alignment of abalone sperm lysin sequences. This dataset is in the file <code>lysin.fna</code>.</p>
<p>Launch HyPhy from the command line by typing <code>HYPHYMP</code>. Navigate through the prompt to reach the FEL analysis: Enter <strong>1</strong> for "Selection Analyses", and then <strong>2</strong> for "FEL". Respond to the following prompts:</p>
<ul>
<li><strong><code>Choose Genetic Code</code></strong>: Enter <code>1</code> to select the Universal genetic code.</li>
<li><strong><code>Select a coding sequence alignment file</code></strong>: Enter the full path to the example dataset, <code>/path/to/tutorial_data/lysin.fna</code></li>
<li><strong><code>A tree was found in the data file:...Would you like to use it (y/n)?</code></strong>: Enter <code>y</code> to use the provided tree.</li>
<li><strong><code>Choose the set of branches to test for selection</code></strong>: To perform tests for diversifying selection that consider all branches, enter option <code>1</code> for <strong>All</strong>. Alternatively, if you wish to test for site-level selection considering only a subset of branches, enter a different option (2,3,4, or other). Note that any labels present in the provided phylogeny will be listed as options in this menu.</li>
<li><strong><code>Use synonymous rate variation? Strongly recommended YES for selection inference.</code></strong>: Enter <code>1</code> to employ synonymous rate variation. If you would like to constrain dS=1 at all sites, for example to calculate evolutionary rate point estimates, enter <code>2</code>. </li>
<li><strong><code>Select the p-value threshold to use when testing for selection</code></strong>: Provide the desired P-value threshold for calling sites as positively selected. We recommend <code>0.1</code> for FEL.</li>
</ul>
<p>FEL will now run to completion and print markdown-formatted status indicators to screen, indicating the progression of model fits and concluding with the final FEL test results:
<img alt="FEL markdown output" src="../files/images/fel-markdown.png" /></p>
<p>Note that FEL will formally test for both positive and negative selection at each site. This analysis found 22 sites under pervasive negative selection and 17 sites under pervasive positive selection at our specified threshold of P&lt;0.1.</p>
<h3 id="use-meme-to-find-sites-which-have-experienced-pervasive-diversifying-selection">Use MEME to find sites which have experienced pervasive diversifying selection.<a class="headerlink" href="#use-meme-to-find-sites-which-have-experienced-pervasive-diversifying-selection" title="Permanent link">#</a></h3>
<blockquote>
<p>See <a href="../../methods/selection-methods/#meme">here</a> for a description of the MEME method.</p>
</blockquote>
<p>We will demonstate MEME use with an alignment of abalone sperm lysin sequences. This dataset is in the file <code>lysin.fna</code>.</p>
<p>Launch HyPhy from the command line by typing <code>HYPHYMP</code>. Navigate through the prompt to reach the MEME analysis: Enter <strong>1</strong> for "Selection Analyses", and then <strong>1</strong> for "MEME". Respond to the following prompts:</p>
<ul>
<li><strong><code>Choose Genetic Code</code></strong>: Enter <code>1</code> to select the Universal genetic code.</li>
<li><strong><code>Select a coding sequence alignment file</code></strong>: Enter the full path to the example dataset, <code>/path/to/tutorial_data/h3_trunk.fna</code></li>
<li><strong><code>A tree was found in the data file:...Would you like to use it (y/n)?</code></strong>: Enter <code>y</code> to use the provided tree.</li>
<li><strong><code>Choose the set of branches to test for selection</code></strong>: To perform tests for episodic selection that consider all branches, enter option <code>1</code> for <strong>All</strong>. Alternatively, if you wish to test for site-level selection considering only a subset of branches, enter a different option (2,3,4, or other). Note that any labels present in the provided phylogeny will be listed as options in this menu.</li>
<li><strong><code>Select the p-value threshold to use when testing for selection</code></strong>: Provide the desired P-value threshold for calling sites as positively selected. We recommend <code>0.1</code> for MEME.</li>
</ul>
<p>MEME will now run to completion and print markdown-formatted status indicators to screen, indicating the progression of model fits and concluding with the final MEME test results):
<img alt="MEME markdown output" src="../files/images/meme-markdown.png" /></p>
<p>Note that MEME will formally test only for positive, but not negative, selection at each site. This analysis found 30 sites under episodic positive selection at our specified threshold of P&lt;0.1.</p>
<h3 id="use-slac-to-find-sites-which-have-experienced-pervasive-diversifying-selection">Use SLAC to find sites which have experienced pervasive diversifying selection.<a class="headerlink" href="#use-slac-to-find-sites-which-have-experienced-pervasive-diversifying-selection" title="Permanent link">#</a></h3>
<blockquote>
<p>See <a href="../../methods/selection-methods/#slac">here</a> for a description of the SLAC method.</p>
</blockquote>
<p>We will demonstate SLAC use with an alignment of abalone sperm lysin sequences. This dataset is in the file <code>lysin.fna</code>.</p>
<p>Launch HyPhy from the command line by typing <code>HYPHYMP</code>. Navigate through the prompt to reach the SLAC analysis: Enter <strong>1</strong> for "Selection Analyses", and then <strong>3</strong> for "SLAC". Respond to the following prompts:</p>
<ul>
<li><strong><code>Choose Genetic Code</code></strong>: Enter <code>1</code> to select the Universal genetic code.</li>
<li><strong><code>Select a coding sequence alignment file</code></strong>: Enter the full path to the example dataset, <code>/path/to/tutorial_data/h3_trunk.fna</code></li>
<li><strong><code>A tree was found in the data file:...Would you like to use it (y/n)?</code></strong>: Enter <code>y</code> to use the provided tree.</li>
<li><strong><code>Choose the set of branches to test for selection</code></strong>: To perform tests for episodic selection that consider all branches, enter option <code>1</code> for <strong>All</strong>. Alternatively, if you wish to test for site-level selection considering only a subset of branches, enter a different option (2,3,4, or other). Note that any labels present in the provided phylogeny will be listed as options in this menu.</li>
<li><strong><code>Select the number of samples used to assess ancestral reconstruction uncertainty</code></strong>: This prompt asks how many bootstrap samples to draw for generating confidence intervals. Provide the value <code>100</code>.</li>
<li><strong><code>Select the p-value threshold to use when testing for selection</code></strong>: Provide the desired P-value threshold for calling sites as positively selected. We recommend <code>0.1</code> for MEME.</li>
</ul>
<p>SLAC will now run to completion and print markdown-formatted status indicators to screen, indicating the progression of model fits and concluding with the final SLAC test results (abbreviated for visual purposes):
<img alt="SLAC markdown output" src="../files/images/slac-markdown.png" /></p>
<p>Note that SLAC will formally test for both positive and negative selection at each site. This analysis found 13 sites under pervasive negative selection and 8 sites under pervasive positive selection at our specified threshold of P&lt;0.1.</p>
<h3 id="use-fubar-to-find-sites-which-have-experienced-pervasive-diversifying-selection">Use FUBAR to find sites which have experienced pervasive diversifying selection.<a class="headerlink" href="#use-fubar-to-find-sites-which-have-experienced-pervasive-diversifying-selection" title="Permanent link">#</a></h3>
<blockquote>
<p>See <a href="../../methods/selection-methods/#fubar">here</a> for a description of the FUBAR method.</p>
</blockquote>
<p>We will demonstate FUBAR use with an alignment of influenza A H3N2 hemagglutinin sequences subsetted from <a href="http://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1004940">Meyer and Wilke 2015</a> to sample sequences only along the trunk. This dataset is in the file <code>h3_trunk.fna</code>.</p>
<ul>
<li><strong><code>Choose Genetic Code</code></strong>: Enter <code>1</code> to select the Universal genetic code.</li>
<li><strong><code>Select a coding sequence alignment file</code></strong>: Enter the full path to the example dataset, <code>/path/to/tutorial_data/h3_trunk.fna</code></li>
<li><strong><code>A tree was found in the data file:...Would you like to use it (y/n)?</code></strong>: Enter <code>y</code> to use the provided tree.</li>
<li><strong><code>Number of grid points per dimension</code></strong>: Press the <code>Enter</code> key to accept the default of 20.</li>
<li><strong><code>Number of MCMC chains to run</code></strong>: Press the <code>Enter</code> key to accept the default of 5.</li>
<li><strong><code>The length of each chain</code></strong>: Press the <code>Enter</code> key to accept the default of 2000000.</li>
<li><strong><code>Use this many samples as burn-in</code></strong>:Press the <code>Enter</code> key to accept the default of 1000000.</li>
<li><strong><code>How many samples should be drawn from each chain</code></strong>: Press the <code>Enter</code> key to accept the default of 100.</li>
<li><strong><code>The concentration parameter of the Dirichlet prior</code></strong>: Press the <code>Enter</code> key to accept the default of 0.5.</li>
</ul>
<p>FUBAR will now run to completion and print markdown-formatted status indicators to screen, indicating the progression of model fits and concluding with the final FUBAR test results (abbreviated output shown here with final result only):
<img alt="FUBAR markdown output" src="../files/images/fubar-markdown.png" /></p>
<p>Note that FUBAR will formally test for both positive and negative selection at each site, although it currently only reports positively selected sites to screen. This analysis found 6 sites under pervasive positive selection at a posterior probability &gt;= 0.9</p>
<h3 id="use-relax-to-compare-selective-pressures-on-different-parts-of-the-tree">Use RELAX to compare selective pressures on different parts of the tree<a class="headerlink" href="#use-relax-to-compare-selective-pressures-on-different-parts-of-the-tree" title="Permanent link">#</a></h3>
<blockquote>
<p>See <a href="../../methods/selection-methods/#relax">here</a> for a description of the RELAX method.</p>
</blockquote>
<p>We will demonstate RELAX use with an alignment of HIV-1 envelope protein sequences collected from  epidemiologically-linked donor-recipient transmission pairs, from <a href="http://jvi.asm.org/content/79/10/6523">Frost et al, 2005</a>. This dataset is in the file <code>hiv1_transmission_labeled.fna</code>.</p>
<p>Launch HyPhy from the command line by typing <code>HYPHYMP</code>. Navigate through the prompt to reach the RELAX analysis: Enter <strong>1</strong> for "Selection Analyses", and then <strong>7</strong> for "RELAX". Respond to the following prompts:</p>
<ul>
<li><strong><code>Choose Genetic Code</code></strong>: Enter <code>1</code> to select the Universal genetic code.</li>
<li><strong><code>Select a coding sequence alignment file</code></strong>: Enter the full path to the example dataset, <code>/path/to/tutorial_data/hiv1_transmission_labeled.fna</code></li>
<li><strong><code>A tree was found in the data file:...Would you like to use it (y/n)?</code></strong>: Enter <code>y</code> to use the provided tree.</li>
<li><strong><code>Choose the set of branches to use as the _test_ set</code></strong>: Enter <code>2</code> to select all branches labeled "test" as the test set. Note that if your tree contains multiple labels, HyPhy will issue a subsequent prompt asking you to specify the reference set of lineages from these labels.</li>
<li><strong><code>RELAX analysis type</code></strong>: Enter <code>1</code> to run only the RELAX test, and <code>2</code> to run the RELAX test as well as fit other descriptive models. Here, enter <code>2</code>.</li>
</ul>
<p>RELAX will now run to completion and print markdown-formatted status indicators to screen, indicating the progression of model fits and concluding with the final RELAX test results (abbreviated output shown here with final result only):
<img alt="RELAX markdown output" src="../files/images/relax-markdown.png" /></p>
<p>This analysis did not detect any evidence of relaxed selection. However, if it had, a significant K&gt;1 would indicate intensified selection on test lineages, and significant K&lt;1 would indicate relaxed selection on test lineages.</p>
          
            <footer class="footer">
              <div class="container citation-footer">
  <p>HyPhy is funded jointly by <a
                 href="https://projectreporter.nih.gov/project_info_details.cfm?aid=9102131"
                 target="_blank">MIDAS</a> and <a
                 href="https://projectreporter.nih.gov/project_info_description.cfm?aid=8886417"
                 target="_blank">NIH award R01 GM093939</a></p>
</div>



            </footer>
          
        </div>
      </article>
      <div class="results" role="status" aria-live="polite">
        <div class="scrollable">
          <div class="wrapper">
            <div class="meta"></div>
            <div class="list"></div>
          </div>
        </div>
      </div>
    </main>
    <script>
      var base_url = '../..';
      var repo_id  = '';
    </script>
    <script src="../../assets/javascripts/application.js"></script>
    
      <script src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML"></script>
    
      <script src="../../assets/mathjaxhelper.js"></script>
    
    
				<!-- Piwik -->
				<script type="text/javascript">
					var _paq = _paq || [];
					_paq.push(["setDomains", ["*.hyphy.org"]]);
					_paq.push(['trackPageView']);
					_paq.push(['enableLinkTracking']);
					(function() {
						var u="//datamonkey.org/piwik/";
						_paq.push(['setTrackerUrl', u+'piwik.php']);
						_paq.push(['setSiteId', 10]);
						var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
						g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'piwik.js'; s.parentNode.insertBefore(g,s);
					})();
				</script>
				<noscript><p><img src="//datamonkey.org/piwik/piwik.php?idsite=10" style="border:0;" alt="" /></p></noscript>
				<!-- End Piwik Code -->
    
    <script charset="utf-8">
      var shiftWindow = function() { scrollBy(0, -65) };
      window.addEventListener("hashchange", shiftWindow);
    </script>
  </body>
</html>